File: solutions/javascript/nth-prime/1/nth-prime.js
//
// This is only a SKELETON file for the 'Nth Prime' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const prime = (n) => {
  if (n == 0)
  throw new Error('there is no zeroth prime');
  if (n == 1) return 2;
  var pr = [2];
  for (var p = 3;; p=p+2) {
    var i;
    for (i = 1; i <= pr.length; i++) {
      if (p % pr[i] == 0) break;
    }
    if (i > pr.length) pr.push(p);
    if (n == pr.length) return p;
  }
};

File: solutions/python/nth-prime/1/nth_prime.py
""" Module docstring """
def prime(number):
    """ function docstring """
    if number == 0:
        raise ValueError('there is no zeroth prime')
    n_pr = [2]
    nxt = 3
    while len(n_pr) < number:
        i = 0
        while n_pr[i] * n_pr[i] < nxt:
            if nxt % n_pr[i] == 0:
                break
            i += 1
        if n_pr[i] * n_pr[i] > nxt:
            n_pr.append(nxt)
        nxt += 2
    return n_pr[-1]
